### DISCUSSION TITLE
[Mailer] Microsoft Outlook SMTP Authentication

### DISCUSSION THREAD

**Discussion by maluramichael** [symfony/symfony#52130] [Q&A]
Hello Community,

we used this string here for sending our mails from the backend `smtp://smtp.office365.com:587?encryption=tls&auth_mode=login&username=OUR_MAIL&password=PASSWORD`

Now changed our admin the authorization settings in office to 2FA and the password authentication does not work anymore.

After searching for answers i tried the following

```php
$transport = new EsmtpTransport('smtp-mail.outlook.com', 587);
$transport->setAuthenticators([
    new XOAuth2Authenticator()
]);
$transport->setUsername('OUR_MAIL');
$transport->setPassword('ACCESS_TOKEN OR PASSWORD???');
$result = $transport->send($message);
```

But this does not seem to work. I still get the error `Expected response code "235" but got code "535", with message "535 5.7.3 Authentication unsuccessful` 

 I think i have to use my client id, secret and tenant to get an access key. But how? Did someone else solved this problem?

Kind regards, Michael

**Comment by Jir4** [comment]
Hi, 

I tried to achieve this in the same way today, i finally got something working using this : https://gist.github.com/dbu/3094d7569aebfc94788b164bd7e59acc
